app/scripts/metamask-controller.js

```diff
@@ -1090,21 +1090,15 @@ export default class MetamaskController extends EventEmitter {
       },
       controllerMessenger: this.controllerMessenger,
     });
-    this.memStore.subscribe(this.sendUpdate.bind(this));
 
     if (
       !this.isUnlocked() &&
       this.onboardingController.store.getState().completedOnboarding
     ) {
-      // Automatic login via config password
-      const password = process.env.CONF?.PASSWORD;
-      if (password) {
-        this.submitPassword(password);
-      }
-      // Automatic login via storage encryption key
-      else if (isManifestV3) {
-        this.submitEncryptionKey();
-      }
+      this._loginUser();
+    } else {
+      this.emit('startUISync');
+      this.memStore.subscribe(this.sendUpdate.bind(this));
```

Comment:
`memStore` updates a lot as controllers initialise, delaying this `subscribe` helps to make sure that incomplete UI state is not flushed to UI.